src/wazuh.py

```diff
@@ -353,3 +355,48 @@ def _generate_cluster_snippet(
             <disabled>no</disabled>
         </cluster>
     """
+
+
+def change_api_password(old_password: str, new_password: str) -> None:
+    """Change Wazuh's API password for the default 'wazuh' user.
+
+    Args:
+        old_password: the old API password for the 'wazuh' user.
+        new_password: the new API password for the 'wazuh' user.
+
+    Raises:
+        WazuhInstallationError: if an error occurs while processing the requests.
+    """
+    # The certificates might be self signed and there's no security hardening in
+    # passing them to the request since tampering with `localhost` would mean the
+    # container filesystem is compromised
+    try:
+        r = requests.get(  # nosec
+            "https://localhost:55000/security/user/authenticate",
+            auth=("wazuh", old_password),
+            timeout=10,
+            verify=False,
+        )
+        # The old password has already been changed. Nothing to do.
+        if r.status_code == 401:
+            return
+        token = r.json()["data"]["token"]
+        r = requests.put(  # nosec
+            "https://localhost:55000/security/users/2",
+            headers={"Authorization": f"Bearer {token}"},
+            data={"password": secrets.token_hex()},
+            timeout=10,
+            verify=False,
+        )
```

Comment:
If the request to change the password for user 2 fails, the subsequent steps (changing the password for user 1) will be skipped. Since the password for user 2 is randomly generated, this makes the function usable only once, even in case of failure. Perhaps it should be split into two `try ... except ...` blocks.

Reply:
Good point